webapp/javascript/pages/adhoc/components/utils.ts

```diff
@@ -0,0 +1,50 @@
+/* eslint-disable default-case, @typescript-eslint/switch-exhaustiveness-check, consistent-return */
```

Comment:
`Utils` is a footgun, if we can let's call it with a more descriptive name.

`labels.ts` or `humanize.ts`

Reply:
Also you don't want to disable `@typescript-eslint/switch-exhaustiveness-check`, since that's what will force us to handle all use cases.

Reply:
Eg
<img width="1132" alt="Screen Shot 2022-08-19 at 12 16 14" src="https://user-images.githubusercontent.com/6951209/185651744-13a62116-1739-4b42-a216-f33e18a1947f.png">

Reply:
Also from what I see we are using `SpyNameFirstClass`, which is a narrower type than `SpyName`. We could export SpyNameFirstClass as an enum type and only deal with the cases we actually care about:

<img width="1209" alt="Screen Shot 2022-08-19 at 12 22 14" src="https://user-images.githubusercontent.com/6951209/185652468-a3754075-d8b2-4a8e-bb17-25d0b71f1808.png">

Reply:
fixed